# This file is part of HiGFXback

# requires
REQUIRES="cmake3-build gdk-pixbuf-build libaom-build libde265-build libstdc++-build libwebp-build x265-build"

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
GENERATOR=${GENERATOR-Unix Makefiles}
PKG_CONFIG_PATH=/x11/lib/pkgconfig cmake -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON -DCMAKE_INSTALL_PREFIX=/x11 -DCMAKE_INSTALL_RPATH=/x11/lib -B build -G "$GENERATOR"

# build
cmake --build build

# install
install -d $DESTDIR/x11/bin
install build/examples/heif-convert $DESTDIR/x11/bin
install build/examples/heif-enc $DESTDIR/x11/bin
install build/examples/heif-info $DESTDIR/x11/bin
install build/examples/heif-thumbnailer $DESTDIR/x11/bin
install -d $DESTDIR/x11/include/libheif
install -m 644 build/libheif/heif_version.h $DESTDIR/x11/include/libheif
install -m 644 libheif/heif.h $DESTDIR/x11/include/libheif
install -d $DESTDIR/x11/lib
install build/libheif/libheif.so.1.* $DESTDIR/x11/lib/libheif.so.1
ln -sf libheif.so.1 $DESTDIR/x11/lib/libheif.so
install -d $DESTDIR/x11/lib/gdk-pixbuf-2.0/2.10.0/loaders
install build/gdk-pixbuf/libpixbufloader-heif.so $DESTDIR/x11/lib/gdk-pixbuf-2.0/2.10.0/loaders
install -d $DESTDIR/x11/lib/pkgconfig
install -m 644 build/libheif.pc $DESTDIR/x11/lib/pkgconfig
test $DESTDIR || /x11/bin/gdk-pixbuf-query-loaders > /x11/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/libheif-build.pc << EOF
Name: libheif
Version: 1.16.2
Description: High Efficiency Image Format (HEIF) library
Requires: $REQUIRES

devel=\\
/x11/include/libheif/heif.h \\
/x11/include/libheif/heif_version.h \\
/x11/lib/libheif.so \\
/x11/lib/pkgconfig/libheif.pc

exec=\\
/x11/bin/heif-convert \\
/x11/bin/heif-enc \\
/x11/bin/heif-info \\
/x11/bin/heif-thumbnailer \\
/x11/lib/libheif.so.1 \\
/x11/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-heif.so
EOF

exit

# patch
--- libheif-1.16.2.orig/cmake/modules/FindJPEG.cmake
+++ libheif-1.16.2/cmake/modules/FindJPEG.cmake
@@ -0,0 +1,30 @@
+include(LibFindMacros)
+
+libfind_pkg_check_modules(JPEG_PKGCONF libjpeg)
+
+find_path(JPEG_INCLUDE_DIR
+    NAMES jpeglib.h
+    HINTS ${JPEG_PKGCONF_INCLUDE_DIRS}
+)
+
+find_library(JPEG_LIBRARY
+    NAMES jpeg
+    HINTS ${JPEG_PKGCONF_LIBRARY_DIRS}
+)
+
+set(JPEG_PROCESS_LIBS ${JPEG_LIBRARY})
+set(JPEG_PROCESS_INCLUDES ${JPEG_INCLUDE_DIR})
+libfind_process(JPEG)
+
+add_library(JPEG::JPEG UNKNOWN IMPORTED)
+set_target_properties(JPEG::JPEG PROPERTIES
+    IMPORTED_LOCATION "${JPEG_LIBRARY}"
+    INTERFACE_INCLUDE_DIRECTORIES "${JPEG_INCLUDE_DIR}"
+)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(JPEG
+    REQUIRED_VARS
+        JPEG_INCLUDE_DIR
+        JPEG_LIBRARIES
+)
--- libheif-1.16.2.orig/cmake/modules/FindPNG.cmake
+++ libheif-1.16.2/cmake/modules/FindPNG.cmake
@@ -0,0 +1,30 @@
+include(LibFindMacros)
+
+libfind_pkg_check_modules(PNG_PKGCONF libpng)
+
+find_path(PNG_INCLUDE_DIR
+    NAMES png.h
+    HINTS ${PNG_PKGCONF_INCLUDE_DIRS}
+)
+
+find_library(PNG_LIBRARY
+    NAMES png
+    HINTS ${PNG_PKGCONF_LIBRARY_DIRS}
+)
+
+set(PNG_PROCESS_LIBS ${PNG_LIBRARY})
+set(PNG_PROCESS_INCLUDES ${PNG_INCLUDE_DIR})
+libfind_process(PNG)
+
+add_library(PNG::PNG UNKNOWN IMPORTED)
+set_target_properties(PNG::PNG PROPERTIES
+    IMPORTED_LOCATION "${PNG_LIBRARY}"
+    INTERFACE_INCLUDE_DIRECTORIES "${PNG_INCLUDE_DIR}"
+)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(PNG
+    REQUIRED_VARS
+        PNG_INCLUDE_DIR
+        PNG_LIBRARIES
+)
--- libheif-1.16.2.orig/gdk-pixbuf/CMakeLists.txt
+++ libheif-1.16.2/gdk-pixbuf/CMakeLists.txt
@@ -3,10 +3,7 @@
   pkg_check_modules(GDKPIXBUF2 gdk-pixbuf-2.0)
 
   if(GDKPIXBUF2_FOUND)
-    execute_process(
-      COMMAND ${PKG_CONFIG_EXECUTABLE} gdk-pixbuf-2.0 --variable gdk_pixbuf_moduledir
-              --define-variable=prefix=${CMAKE_INSTALL_PREFIX} OUTPUT_VARIABLE GDKPIXBUF2_MODULE_DIR
-      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    pkg_get_variable(GDKPIXBUF2_MODULE_DIR gdk-pixbuf-2.0 gdk_pixbuf_moduledir)
 
     add_library(pixbufloader-heif MODULE pixbufloader-heif.c)
 
# source
https://github.com/strukturag/libheif/archive/v1.16.2/libheif-1.16.2.tar.gz
